### ISSUE TITLE
[v1a1->v1a2] incomplete resources causes error

### ISSUE THREAD

**Issue by anoipm** [kyma-project/kyma#15012]
After conversion function CRD with incomplete definition of resources we have error during running (deploying?) it.

input:
```
apiVersion: serverless.kyma-project.io/v1alpha1
kind: Function
metadata:
  name: function-008-1
  namespace: default
spec:
  resources:
    limits:
      cpu: 1
  runtime: python39
  source: |-
    def main(event, context):
        return "hello world"
```
result (without `status`!):
```
apiVersion: serverless.kyma-project.io/v1alpha2
kind: Function
metadata:
  name: function-008-1
  [...]
spec:
  resourceConfiguration:
    build:
      resources: {}
    function:
      resources:
        limits:
          cpu: "1"
  runtime: python39
  source:
    inline:
      dependencies: ""
      source: |-
        def main(event, context):
            return "hello world"
  template: {}
status: {}
```

`kubectl logs -n kyma-system serverless-ctrl-mngr-cc5fddc8f-crbx5 | grep function-008-1` shows repeatedly:
```
{"level":"WARN","timestamp":"2022-08-04T08:07:00Z","logger":"controllers.function","caller":"serverless/fsm.go:135","message":"while updating function status: admission webhook \"validation.webhook.serverless.kyma-project.io\" denied the request: validation failed: [invalid function resources: [spec.resourceConfiguration.function.resources.limits.memory(0) should be higher than minimal value (16Mi)]]","context":{"kind":"Function","name":"function-008-1","namespace":"default","version":1}}
```

I am using kyma taken from main on August 2.
For conversion I use command `kubectl apply -f input.yaml`.

**Comment by anoipm** [comment]
It's probably connected with #15003 because the same configuration with `v1alpha2` works correctly.

input:
```
apiVersion: serverless.kyma-project.io/v1alpha2
kind: Function
metadata:
  name: function-008-2
  namespace: default
spec:
  resourceConfiguration:
    function:
      resources:
        limits:
          cpu: 1
  runtime: python39
  source:
    inline:
      dependencies: ""
      source: |-
        def main(event, context):
            return "hello world"
```
result (with default resources and status is running):
```
apiVersion: serverless.kyma-project.io/v1alpha2
kind: Function
metadata:
  name: function-008-2
  [...]
spec:
  resourceConfiguration:
    build:
      resources: {}
    function:
      resources:
        limits:
          cpu: "1"
          memory: 128Mi
        requests:
          cpu: 50m
          memory: 64Mi
  [...]
status:
  conditions:
  - lastTransitionTime: "2022-08-04T08:10:01Z"
    message: Deployment function-008-2-ntc2q is ready
    reason: DeploymentReady
    status: "True"
    type: Running
  [...]
```

**Comment by kwiatekus** [comment]
Fixed with #15049
